### PR TITLE
Package PR build trigger

### PR DESCRIPTION
**Pull request by Igor-Techsee**

### CODE REVIEW

File: .github/workflows/npm-build-common.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+name: Build package reusable task
```
Comment (royricht) [inline .github/workflows/npm-build-common.yml:1]: Set appropriate name

File: .github/workflows/npm-build-common.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+on:
+    # Testing reusable workflows
```
Comment (royricht) [inline .github/workflows/npm-build-common.yml:4]: Appropriate name

File: .github/workflows/npm-build-common.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    workflow_call:
+        # Inputs the workflow accepts.
+        inputs:
+            branch:
+                required: true
+                type: string
+            use-slack:
+                description: 'Should there be a slack notification'
+                type: boolean
+                default: false
```
Comment (royricht) [inline .github/workflows/npm-build-common.yml:14]: Should we not default to true? Its going to be like this anyways

File: .github/workflows/npm-build-common.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            package-name:
+                description: 'A package name to be used in the slack notification'
+                type: string
+                required: false
```
Comment (royricht) [inline .github/workflows/npm-build-common.yml:18]: I think should be true

File: .github/workflows/npm-build-common.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        secrets:
+            git-token:
+                description: 'A git token passed from the caller workflow'
+                required: true
+            npm-token:
+                description: 'A npm token passed from the caller workflow'
+                required: true
+            slack-webhookurl:
+                description: 'A slack webhook url for status notification'
+                required: false
```
Comment (royricht) [inline .github/workflows/npm-build-common.yml:29]: Wonder if we can enforce this when slack is enabled for flow

File: .github/workflows/npm-build-common.yml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                  npm run build
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.npm-token }}
+            - name: Post success to Slack
+              if: ${{ success() && inputs.use-slack }}
+              uses: slackapi/slack-github-action@v1.24.0
+              with:
+                  payload: |
+                      {
+                          "package-name": "${{ inputs.package-name }}",
```
Comment (royricht) [inline .github/workflows/npm-build-common.yml:67]: Cant we use the url instead?

Comment (Igor-Techsee) [reply .github/workflows/npm-build-common.yml:67]: the url composed in the slack workflow

**Review by royricht — APPROVED** [review]

**Pull request merged**